src/core/dynamic_batch_scheduler.cc

```diff
@@ -264,22 +336,27 @@ DynamicBatchScheduler::SchedulerThread(
         // Debugging/testing... wait until queue contains 'delay_cnt'
         // items...
         wait_microseconds = 10 * 1000;
-        if (queue_.size() >= delay_cnt) {
+        if (queue_.Size() >= delay_cnt) {
           delay_cnt = 0;
         }
         LOG_INFO << "Delaying scheduler thread " << runner_id << " until "
                  << delay_cnt
-                 << " queued payloads, current total = " << queue_.size();
-      } else if (queue_.empty()) {
+                 << " queued payloads, current total = " << queue_.Size();
+      } else if (queue_.Empty()) {
         wait_microseconds = default_wait_microseconds;
       } else if (dynamic_batching_enabled_) {
         // Use dynamic batching to get request payload(s) to execute.
         wait_microseconds = GetDynamicBatch(runner_id);
-        if (wait_microseconds == 0) {
+
+        // Get payloads that are rejected from searching dynamic batch.
+        rejected_payloads = queue_.ReleaseRejectedPayloads();
```

Comment:
This is copying a vector of payloads... should instead use a shared_ptr or unique_ptr for efficiency.